src/org/org.ts

```diff
@@ -698,18 +700,21 @@ export class Org extends AsyncOptionalCreatable<Org.Options> {
     const globalInfo = await GlobalInfo.getInstance();
     this.logger = await Logger.child('Org');
 
-    this.configAggregator = this.options.aggregator ? this.options.aggregator : await ConfigAggregator.create();
+    this.configAggregator = this.options.aggregator ?? (await ConfigAggregator.create());
 
     if (!this.options.connection) {
-      if (this.options.aliasOrUsername == null) {
-        this.configAggregator = this.getConfigAggregator();
-        const aliasOrUsername = this.options.isDevHub
-          ? this.configAggregator.getPropertyValue<string>(OrgConfigProperties.TARGET_DEV_HUB)
-          : this.configAggregator.getPropertyValue<string>(OrgConfigProperties.TARGET_ORG);
-        this.options.aliasOrUsername = aliasOrUsername || undefined;
+      this.options.aliasOrUsername ??= this.options.isDevHub
+        ? // prefer the sf config but fall back to the sfdx ones
+          this.configAggregator.getPropertyValue<string>(OrgConfigProperties.TARGET_DEV_HUB) ??
+          this.configAggregator.getPropertyValue<string>(SfdxPropertyKeys.DEFAULT_DEV_HUB_USERNAME)
+        : this.configAggregator.getPropertyValue<string>(OrgConfigProperties.TARGET_ORG) ??
+          this.configAggregator.getPropertyValue<string>(SfdxPropertyKeys.DEFAULT_USERNAME);
+
+      if (!this.options.aliasOrUsername) {
+        throw messages.createError('usernameRequired');
       }
 
-      const username = globalInfo.aliases.resolveUsername(this.options.aliasOrUsername as string);
+      const username = globalInfo.aliases.resolveUsername(this.options.aliasOrUsername);
```

Comment:
type assertion is no longer needed because of the above check.  I'd like to avoid dishonest assertions wherever possible.  It's almost always a sign that some check is missing.